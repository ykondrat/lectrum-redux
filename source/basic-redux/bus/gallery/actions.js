import {
    SHOW_NEXT_PHOTO,
    SHOW_PREV_PHOTO,
    SHOW_PHOTO
} from './types';

export const showNextPhoto = () => ({
    type: SHOW_NEXT_PHOTO
});

export const showPrevPhoto = () => ({
    type: SHOW_PREV_PHOTO
});

export const showPhoto = (value) => ({
    type: SHOW_PHOTO,
    payload: value
});
