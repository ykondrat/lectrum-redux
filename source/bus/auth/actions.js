// Types
import { types } from './types';

// Async
export const signupAsync = (userData) => ({
    type: types.SIGNUP_ASYNC,
    payload: userData
});

export const loginAsync = (credentials) => ({
    type: types.LOGIN_ASYNC,
    payload: credentials
});

export const authenticateAsync = (credentials) => ({
    type: types.AUTHENTICATE_ASYNC,
    payload: credentials
});

export const initializeAsync = () => ({
    type: types.INITIALIZE_ASYNC
});

export const logoutAsync = () => ({
    type: types.LOGOUT_ASYNC
});

// Sync
export const authenticate = () => ({
    type: types.AUTHENTICATE
});

export const initialize = () => ({
    type: types.INITIALIZE
});

export const logout = () => ({
    type: types.LOGOUT
});
