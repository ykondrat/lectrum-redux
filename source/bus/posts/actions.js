// Types
import { types } from './types';

// Async
export const fetchPostsAsync = () => ({
    type: types.FETCH_POSTS_ASYNC
});

// Sync
export const fillPosts = (posts) => ({
    type: types.FILL_POSTS,
    payload: posts
});
