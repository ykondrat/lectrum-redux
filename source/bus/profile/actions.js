// Types
import { types } from './types';

// Async


// Sync
export const clearProfile = () => ({
    type: types.CLEAR_PROFILE
});

export const fillProfile = (profile) => ({
    type: types.FILL_PROFILE,
    payload: profile
});
