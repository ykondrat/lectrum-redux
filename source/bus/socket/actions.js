import { socket } from '../../init/socket';
import { api } from '../../API';
import {
    setOnlineState,
    setOfflineState
} from '../ui/actions';
import {
    createPost,
    removePost,
    likePost,
    unlikePost
} from '../posts/actions';
import { Map } from 'immutable';

export const listenConnection = () => (dispatch) => {
    socket.on('connect', () => {
        dispatch(setOnlineState());
    });

    socket.on('disconnect', () => {
        dispatch(setOfflineState());
    });
};

export const listenPosts = () => (dispatch, getState) => {
    socket.on('like', async (event) => {
        const response = JSON.parse(event);

        if (response.meta.action === 'like') {
            const { firstName, lastName } = await api.users.getById.call(api, response.data.userId);
            const liker = Map({
                firstName,
                lastName,
                id: response.data.userId
            })
            dispatch(likePost({
                postId: response.data.postId,
                liker
            }));
        } else {
            dispatch(unlikePost(response.data));
        }
    });

    socket.on('create', (event) => {
        const response = JSON.parse(event);
        dispatch(createPost(response.data));
    });

    socket.on('remove', (event) => {
        const response = JSON.parse(event);
        dispatch(removePost(response.data));
    });
}
