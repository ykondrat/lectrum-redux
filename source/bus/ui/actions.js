// Types
import { types } from './types';

// Sync
export const startFetching = () => ({
    type: types.START_FETCHING
});

export const stopFetching = () => ({
    type: types.STOP_FETCHING
});
