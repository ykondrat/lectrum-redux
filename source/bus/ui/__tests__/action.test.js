import * as actions from "../actions";

describe("ui actions tests:", () => {
  test("startFetching test", () => {
    expect(actions.startFetching()).toMatchSnapshot();
  });
  test("setOnlineState test", () => {
    expect(actions.setOnlineState()).toMatchSnapshot();
  });
  test("setOfflineState test", () => {
    expect(actions.setOfflineState()).toMatchSnapshot();
  });
  test("stopFetching test", () => {
    expect(actions.stopFetching()).toMatchInlineSnapshot(`
Object {
  "type": "STOP_FETCHING",
}
`);
  });
});
