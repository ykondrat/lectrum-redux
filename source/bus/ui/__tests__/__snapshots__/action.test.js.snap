// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui actions tests: setOfflineState test 1`] = `
Object {
  "type": "SET_OFFLINE_STATE",
}
`;

exports[`ui actions tests: setOnlineState test 1`] = `
Object {
  "type": "SET_ONLINE_STATE",
}
`;

exports[`ui actions tests: startFetching test 1`] = `
Object {
  "type": "START_FETCHING",
}
`;
